### PR TITLE
Enable Signed Certificate creation

### PR DESCRIPTION
**Pull request by dciborow**
By setting the issuerParameters.name parameter, we can enable creating a signed certificate.

## Description

If you haven't already, read the full [contribution guide](../CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](../CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [x] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.

### CODE REVIEW

File: modules/deployment-scripts/create-kv-certificate/main.bicep [hunk 1]
```diff
@@ -121,7 +128,12 @@ resource createImportCert 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
       until [ $retryLoopCount -ge $retryMax ]
       do
         echo "Creating AKV Cert $certName with CN $certCommonName (attempt $retryLoopCount)..."
-        az keyvault certificate create --vault-name $akvName -n $certName -p "$(az keyvault certificate get-default-policy | sed -e s/CN=CLIGetDefaultPolicy/CN=${certCommonName}/g )" \
+        az keyvault certificate create \
+          --vault-name $akvName \
+          -n $certName \
+          -p "$(az keyvault certificate get-default-policy
```
Comment (dciborow) [inline modules/deployment-scripts/create-kv-certificate/main.bicep:134]:
```suggestion
          -p "$(az keyvault certificate get-default-policy \
```

**Comment by shenglol** [comment]
@Gordonby do you mind taking a look at the changes?

**Comment by Gordonby** [comment]
> @Gordonby do you mind taking a look at the changes?

Been delayed... but not forgotten 😼

**Comment by Gordonby** [comment]
I think this needs a test added to the main.test.bicep and an example of doing this to the readme.

I ran this, and it failed... I'm assuming there's a prerequisite.  Could this capability be illustrated clearer for the repo users?

```bicep
module akvCertIssuer '../main.bicep' = {
  name: 'akvCertAltIssuer'
  params: {
    akvName:  akv.name
    location: location
    certificateName: 'certIssueC'
    initialScriptDelay: '0'
    issuerName: 'aDifferentIssuer'
  }
}
```

File: modules/deployment-scripts/create-kv-certificate/main.bicep [hunk 2]
```diff
@@ -39,6 +39,9 @@ param initialScriptDelay string = '0'
 @description('When the script resource is cleaned up')
 param cleanupPreference string = 'OnSuccess'
 
+@description('Unknown, Self, or {IssuerName} for certifcate signing')
```
Comment (dciborow) [inline modules/deployment-scripts/create-kv-certificate/main.bicep:42]: sp

**Comment by dciborow** [comment]
@Gordonby, it looks like I missed porting some changes. Everything is here now. 

Going to also go register this sub in our internal Cert Manager so that we can run the full test.

File: modules/deployment-scripts/create-kv-certificate/main.bicep [hunk 3]
```diff
@@ -39,7 +39,11 @@ param initialScriptDelay string = '0'
 @description('When the script resource is cleaned up')
 param cleanupPreference string = 'OnSuccess'
 
-var neededRBACRoles = !empty(rbacRolesNeededOnKV) ? rbacRolesNeededOnKV : 'a4417e6f-fecd-4de8-b567-7b0420556985'
+@description('Unknown, Self, or {IssuerName} for certificate signing')
+param issuerName string = 'Self'
+
+@description('Certificate Issuer Provider')
```
Comment (dciborow) [inline modules/deployment-scripts/create-kv-certificate/main.bicep:45]:
```suggestion
@description('Certificate Issuer Provider, public options include DigiCert and GlobalSign. Additional private options may be available.')
```

**Comment by Gordonby** [comment]
I'll do some regression testing once the `Deployment Test - CI` check is passing 😼

**Pull request closed without merging**